activerecord/lib/active_record/schema_dumper.rb

```diff
@@ -211,13 +211,9 @@ def index_parts(index)
           index.columns.inspect,
           "name: #{index.name.inspect}",
         ]
-        index_parts << 'unique: true' if index.unique
-
-        index_lengths = (index.lengths || []).compact
-        index_parts << "length: #{Hash[index.columns.zip(index.lengths)].inspect}" if index_lengths.any?
```

Comment:
https://github.com/rails/rails/pull/26785#issuecomment-254106539
> Originally `(index.lengths || []).compact` have a bug.
https://github.com/rails/rails/blob/5-0-stable/activerecord/lib/active_record/schema_dumper.rb#L216
If #26785 will backport, it is better to backport #26745 as well.

If adding `t.index [:firm_id, :type], name: "company_partial_index", length: { type: 10 }`, it will be dumped as `t.index ["firm_id", "type"], name: "company_partial_index", length: {"firm_id"=>10}`.